zarr/v3/abc/store.py

```diff
@@ -1,20 +1,19 @@
 from abc import abstractmethod, ABC
 
-from typing import List, Tuple
+from typing import List, Tuple, Optional
 
 
 class Store(ABC):
-    pass
-
-
-class ReadStore(Store):
     @abstractmethod
-    async def get(self, key: str) -> bytes:
+    async def get(
+        self, key: str, byte_range: Optional[Tuple[int, Optional[int]]] = None
+    ) -> Optional[bytes]:
```

Comment:
Question for discussion. Do we want to have the `byte_range` parameter in the single key get method or just in `get_partial_values`? I lean toward this API but it is somewhat duplicative and goes against the suggestion in the [spec](https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html#abstract-store-interface):

```
get - Retrieve the value associated with a given key.

Parameters: key
Output: value
```

Reply:
My instinct would be to ignore the abstract store interface and just do what makes sense locally.

I don't understand why the spec describes both `get(key) -> value` _and_ `get_partial_values(key_range) -> List[Optional[value]]`, as opposed to `get(key, range) -> Optional[value]` and letting clients handle iteration themselves.

Reply:
Thinking about this more, we are definitely going to want to be able to pass the store a batch of values keys/ranges to the store. This will let us push optimizations down to lower level code or let us do optimizations like coalescing requests.